### ISSUE TITLE
Ordered list nesting behaves unexpectedly

### ISSUE THREAD

**Issue by lexthanthree** [dendronhq/dendron#2927]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [ ] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [x] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

Create an ordered list (indentation is 2 spaces)
```markdown
1. Thing 1
  1. Thing 2 
```

Notice that the list items are siblings instead of parent/child

## To Reproduce

See above

## Expected behavior

Thing 2 should be a child of Thing 1

## Desktop (please complete the following information)

- OS: Macos
- Versions (all that apply to bug):0.94 
  - Dendron Extension:  0.94
  - VS Code version: 0.94
  - Dendron CLI: 0.94

**Comment by jonathanyeung** [comment]
Thanks @lexthanthree for reporting the issue. 

For what it's worth, VS Code's built in markdown preview behaves the same way, but I agree that both 2-space and 4-space indent should create a parent/child.

**Comment by lexthanthree** [comment]
@jonathanyeung yeah its odd - it may be some weird spec oversite even in MD? i don't know why they would enforce this with OL and not UL it seems like inconsistent behavior to me
